### PR TITLE
Add file pruning test for predicate using primitive fields in ROW column in Delta Lake Connector

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Relates to https://github.com/trinodb/trino/issues/17159

### CODE REVIEW

**Review by raunaqmorarka — APPROVED** [review]

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java [hunk 1]
```diff
@@ -358,6 +359,45 @@ public void testParquetStatisticsPruning()
         testCountQuery("SELECT count(*) FROM parquet_struct_statistics WHERE row = ROW(2, 'b')", 3, 9);
     }
 
+    @Test
+    public void testPrimitiveFieldsInsideRowColumnPruning()
+    {
+        // Data generated using:
+        // CREATE TABLE nested_fields (id INT, parent STRUCT<child1: BIGINT, child2: STRING>, grandparent STRUCT<parent1 STRUCT<child1: DOUBLE, child2: BOOLEAN>, parent2 STRUCT<child1: DATE>>) USING DELTA LOCATION 's3://krvikash-test/databricks/nested_fields';
+        // INSERT INTO nested_fields VALUES (1, struct('100', 'INDIA'), struct(struct(10.99, true), struct('2023-01-01'))),
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:367]: nit: Wrap before `(1, s ...`.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java [hunk 2]
```diff
@@ -358,6 +359,45 @@ public void testParquetStatisticsPruning()
         testCountQuery("SELECT count(*) FROM parquet_struct_statistics WHERE row = ROW(2, 'b')", 3, 9);
     }
 
+    @Test
+    public void testPrimitiveFieldsInsideRowColumnPruning()
+    {
+        // Data generated using:
+        // CREATE TABLE nested_fields (id INT, parent STRUCT<child1: BIGINT, child2: STRING>, grandparent STRUCT<parent1 STRUCT<child1: DOUBLE, child2: BOOLEAN>, parent2 STRUCT<child1: DATE>>) USING DELTA LOCATION 's3://krvikash-test/databricks/nested_fields';
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:366]: nit: Remove `LOCATION ...`.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java [hunk 3]
```diff
@@ -358,6 +359,45 @@ public void testParquetStatisticsPruning()
...
+        //    (5, struct('500', 'JAPAN'), struct(struct(50.99, true), struct('2023-05-01'))),
+        //    (6, struct('600', 'UK'), struct(struct(60.99, false), struct('2023-06-01'))),
+        //    (7, struct('700', 'FRANCE'), struct(struct(70.99, true), struct('2023-07-01'))),
+        //    (8, struct('800', 'BRAZIL'), struct(struct(80.99, false), struct('2023-08-01'))),
+        //    (9, struct('900', 'NAMIBIA'), struct(struct(90.99, true), struct('2023-09-01'))),
+        //    (10, struct('1000', 'RSA'), struct(struct(100.99, false), struct('2023-10-01')));
+
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE id > 6",
+                Set.of(90.99, 80.99, 70.99, 100.99),
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:380]: nit: Ordering by asc (`70.99, 80.99, 90.99, 100.99` ) will improve the readability.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java [hunk 4]
```diff
@@ -358,6 +359,45 @@ public void testParquetStatisticsPruning()
...
+
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE id > 10",
+                Set.of(),
+                0);
+
+        // pruning does not work on primitive fields inside a struct, expected splits should be 1 after file pruning
+        assertResultAndSplitCount(
+                "SELECT grandparent.parent1.child1 FROM nested_fields WHERE parent.child1 > 600",
+                Set.of(90.99, 80.99, 70.99, 100.99),
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:391]: nit: Ordering by asc (`70.99, 80.99, 90.99, 100.99` ) will improve the readability.

**Review by ebyhr — APPROVED** [review]

**Comment by krvikash** [comment]
Addressed comments.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java [hunk 5]
```diff
@@ -358,6 +359,33 @@ public void testParquetStatisticsPruning()
         testCountQuery("SELECT count(*) FROM parquet_struct_statistics WHERE row = ROW(2, 'b')", 3, 9);
     }
 
+    @Test
+    // See README.md for table values
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:363]: I don't think code readers want to look for README file when seeing this test.

Comment (krvikash) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java:363]: I have copied it as https://github.com/trinodb/trino/pull/17160/files/7695b9a72e0122cc99bffe66ed47a03482a205da#diff-ec00db9ee4bb8d02ce63d04b35bf788c311afef51b7c56bdf23ab5ad8dc5101eR341

Okat

Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestSplitPruning.java]: @krvikash I meant I didn't want code readers waste their time to look for the README file. When we see **_only_** `testPrimitiveFieldsInsideRowColumnPruning` test, we can't understand why these values are expected. 

Anyway, I'm going to merge this PR as-is.

**Pull request merged**